packages/gtmcore/gtmcore/inventory/inventory.py

```
@@ -83,6 +83,44 @@ def _put_labbook(self, path: str, username: str, owner: str) -> LabBook:
         lb = self.load_labbook_from_directory(final_path)
         return lb
 
+    @staticmethod
+    def update_linked_dataset(labbook: LabBook, username: str, init: bool = False) -> None:
+        """
+
+        Args:
+            labbook:
+            username:
+            init:
+
+        Returns:
+
+        """
+        if len(labbook.git.repo.submodules) > 0:
+            for submodule in labbook.git.list_submodules():
+                try:
+                    namespace, dataset_name = submodule['name'].split("&")
+                    rel_submodule_dir = os.path.join('.gigantum', 'datasets', namespace, dataset_name)
+                    submodule_dir = os.path.join(labbook.root_dir, rel_submodule_dir)
+
+                    if init:
+                        # Optionally Init submodule
+                        call_subprocess(['git', 'submodule', 'init', rel_submodule_dir],
+                                        cwd=labbook.root_dir, check=True)
+                    # Update submodule
+                    call_subprocess(['git', 'submodule', 'update', rel_submodule_dir],
+                                    cwd=labbook.root_dir, check=True)
+
+                    ds = InventoryManager().load_dataset_from_directory(submodule_dir)
+                    ds.namespace = namespace
+                    manifest = Manifest(ds, username)
+                    manifest.link_revision()
+
+                except Exception as err:
+                    logger.error(f"Failed to import submodule: {submodule['name']}. This may be an actual error or"
```

This error message should explain that there was a problem with initializing or updating a dataset (user-facing vocabulary). Could additionally explain that it's a git submodule, but doesn't seem necessary

---

this is logged, not sent to the user. These errors are ignored because they will fail if you don't have permissions. I did clean the message up a bit though.